src/execution/vm/bytecode_generator.cpp

```diff
@@ -1833,13 +1842,13 @@ void BytecodeGenerator::VisitBuiltinArithmeticCall(ast::CallExpr *call, ast::Bui
 
   switch (builtin) {
     case ast::Builtin::Abs: {
-      LocalVar src = VisitExpressionForRValue(call->Arguments()[0]);
+      LocalVar src = VisitExpressionForSQLValue(call->Arguments()[0]);
```

Comment:
Change the trailing call to `GetExecutionResult()->SetDestination(dest);` (i.e., remove `dest.ValueOf()`. That should fix your nesting problem.

Reply:
That worked, thanks!